terraform/vsphere-virtual-machine/content-library-ovf-hcp-packer/main.tf

```diff
@@ -0,0 +1,77 @@
+provider "vsphere" {
+  vsphere_server       = var.vsphere_server
+  user                 = var.vsphere_username
+  password             = var.vsphere_password
+  allow_unverified_ssl = var.vsphere_insecure
+}
+
+data "vsphere_datacenter" "datacenter" {
+  name = var.vsphere_datacenter
+}
+
+data "vsphere_network" "network" {
+  name          = var.vsphere_network
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_compute_cluster" "cluster" {
+  name          = var.vsphere_cluster
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_resource_pool" "pool" {
+  name          = format("%s%s", data.vsphere_compute_cluster.cluster.name, "/Resources")
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "vsphere_datastore" "datastore" {
+  name          = var.vsphere_datastore
+  datacenter_id = data.vsphere_datacenter.datacenter.id
+}
+
+data "hcp_packer_image" "image" {
+  bucket_name    = var.hcp_packer_bucket_name
+  cloud_provider = "vsphere"
+  channel        = var.hcp_packer_channel
+  region         = var.vsphere_datacenter
+}
+
+data "vsphere_content_library" "content_library" {
+  name = split("/", data.hcp_packer_image.image.labels.content_library_destination)[0]
+}
+
+data "vsphere_content_library_item" "content_library_item" {
+  name       = split("/", data.hcp_packer_image.image.labels.content_library_destination)[1]
+  type       = "ovf"
+  library_id = data.vsphere_content_library.content_library.id
+}
+
+resource "vsphere_virtual_machine" "vm" {
+  name                    = var.vm_name
+  folder                  = var.vsphere_folder
+  num_cpus                = var.vm_cpus
+  memory                  = var.vm_memory
+  firmware                = var.vm_firmware
+  efi_secure_boot_enabled = var.vm_efi_secure_boot_enabled
+  datastore_id            = data.vsphere_datastore.datastore.id
+  resource_pool_id        = data.vsphere_resource_pool.pool.id
+  network_interface {
+    network_id = data.vsphere_network.network.id
+  }
+  disk {
+    label            = "disk0"
+    size             = var.vm_disk_size
+    thin_provisioned = true
+  }
+  clone {
+    template_uuid = data.vsphere_content_library_item.content_library_item.id
+    customize {
+      linux_options {
```

Comment:
May need to set the example directory name to Linux specific.

Reply:
Done in commit 769325209ff3b4b6ae1d93e2262e7b32b449bf4d